added docs for insecure, etcd-metrics, inlinemanifests and extraManifests (#11430)
siderolabs/talos: website/content/v1.10/talos-guides/configuration/insecure.md

---
title: "The insecure flag"
description: "Learn how to use the insecure flag."
---

> The `--insecure` flag is a per-command argument that allows you to communicate with the Talos API when a node is in maintenance mode, that is, before a node has been configured with a machine configuration or TLS certificates.

Reviewer: This gives the wrong impression about the security.

`--insecure` in this context means "don't verify the server's TLS certificate". Even in maintenance mode API runs over TLS, but the identity of both client and server are not verified.

Probably this section should start with a note that regular Talos API uses mutual TLS verification, while in maintenance mode client doesn't present any certificate, while Talos (server) uses self-signed TLS certificate, which `--insecure` mode skips a check for.

Without these credentials, nodes in this maintenance state can only accept API calls through the insecure interface.

Only a small subset of Talos API commands support the `--insecure` flag, specifically those required for initial setup and maintenance operations.
However, once you've applied a machine config, you must stop using the `--insecure` flag for all subsequent operations.
The node will now expect secure communication through a talosconfig file.

**Note**: The `--insecure` flag is used in a different context by the `talosctl image cache-create` command.
This command is not used for interacting with the Talos node, but for allowing access to insecure image registries that do not support TLS.

## Supported Commands With the insecure Flag

The following commands can be used with the `--insecure` flag:

`talosctl apply-config`

Use this command alongside the `--insecure` flag to apply a machine configuration for the first time.

`talosctl version`

Check the Talos version running on the node.

`talosctl get`

Retrieves resources from the node.

> **Note**: You **cannot** retrieve the following resources in insecure mode:

Reviewer: I think this list will be always outdated, so it doesn't make any sense to include it in the docs.

`talosctl get rd` gives a list of resource definitions, those which have `sensitive: true` are not available

Reviewer 2: Is there a way to filter based on that property? This is a common question we get in slack and it would be helpful if we could do something like
```
talosctl get rd --insecure
```
to see resources that are available in insecure mode. Otherwise we could write a quick for loop that gets them.

Reviewer: You can do `talosctl get rd --insecure`, but it means getting resource definitions in insecure mode. You can do `-o yaml` and filter with `yq`, but I don't think anybody needs this for real in the docs. It will be outdated anyways. If they hit a resource which is not available, it's not available, and the message is clear about it.

* The client (talosctl) does not present a certificate.
* Neither side can verify the other's identity.

In this case, the `--insecure` flag tells `talosctl` to skip verifying the server’s certificate, allowing the connection to proceed.

Only a small subset of Talos API commands support the --insecure flag, specifically those required for initial setup and maintenance operations.

However, once you've applied a machine config, you must stop using the `--insecure` flag for all subsequent operations.
The node will now expect secure communication through a talosconfig file.

**Note**: The `--insecure` flag is used in a different context by the `talosctl image cache-create` command.
This command is not used for interacting with the Talos node, but for allowing access to insecure image registries that do not support TLS.

## In Omni-Managed Clusters

The `--insecure` flag works differently when you're using Omni to manage Talos clusters.

Here, the flag is used for nodes that haven't joined a cluster yet.
These nodes will only listen for communication over the SideroLink connection, a secure VPN point-to-point connection between Omni and the Talos node.

> So the SideroLink connection is the only way you can run commands against a node in insecure mode.

Reviewer 2: ```suggestion
So the SideroLink connection is the only way you can run commands against a node connected to Omni.
```

## Supported Commands With the insecure Flag

The following commands can be used with the --insecure flag:

`talosctl apply-config`

Use this command alongside the `--insecure` flag to apply a machine configuration for the first time.

`talosctl version`

Check the Talos version running on the node.

`talosctl get`

Retrieves resources from the node.
Verify which resources are retrievable in `--insecure` mode by following these steps:

1. Set your Talos node IP address as a variable (replace <node_ip> with the IP address of your Talos node):

    ```bash
    NODE_IP=<node_ip>

    ```

1. List resources available in `--insecure` mode:

    ```bash
    talosctl get rd --insecure --nodes $NODE_IP -o json \
    | jq -r 'select(.spec.sensitivity == null) | .spec.aliases[0]'

    ```

1. List resources not available in `--insecure` mode:

    ```bash
    talosctl get rd --insecure --nodes $NODE_IP -o json \
    | jq -r 'select(.spec.sensitivity != null) | .spec.aliases[0]'

    ```

`talosctl meta`

Manages key-value pairs in the META partition.

`talosctl reset`

Resets the nodes in Omni.

> `talosctl upgrade`

Reviewer: drop it

Upgrades the Talos versions  in the nodes in Omni.

`talosctl wipe disk`

Erase data from disk partitions on a Talos node.

> Refer to the [CLI reference](https://www.talos.dev/v1.10/reference/cli/) for full CLI details.

Author: My apologies, I added it as an external link in my Google docs, so when I exported as markdown I missed that link
